### PR TITLE
Life Proc Changes--Cold Space and and Temp Handling

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Cleans up the life proc and makes temperature handle better (in preparation for LINDA, partially, and also for better balance/experience in general).

- Pressure and temperature are handled without as many calculations; less fluff and stuff, more down to the basics.
- Breathing temperature damage and body temperature damage are two very distinctly separate damage points now for the 3rd damage level (longstanding Bay oversight).
 - Space is cold, once again. A firesuit+helmet will give you pressure immunity, but it won't protect you against the cold. 
 - This does NOT mean that opening a door to space will make the room cold; this only makes space turfs cold.
 - Tajaran and Unathi's temperature benefits/downsides are now a simple +20/-20 strength/weakness to temperatures.

Quirks: 
- Bodytemps stabilize at 76F, like they do on TG and Goon instead of the current 98F.
 - Virus fevers aren't noticeable in terms of causing burn damage (still can easily notice fevers on the health analyzer due to elevated body temperature).
 - Upside to this is cryo probably won't be so infuriating for it to lower body temps to a level where it will start healing them.

Fixes:
- Fixes frost oil so it actually functions properly.
- Fixes issues with not being able to proc level 3 fire damage when you're in an actual fire

Other Stuff: 
- Removes some of the more annoying effects of plasma (eye burning, clothes contamination, etc), but lowers the air concentration level required for plasma gas to cause toxin damage.

- Fire/hotspots no longer cause extra damage just because its fire; pushes damage from fires entirely onto the mob itself, based on temperature
 - body temp damage levels 2 and 3 lowered for machine people to compensate for this proc being removed.

- N2O gas concentrations required to trigger giggling and laughing lowered.

- Removes the unused (and now unnecessary) suit cooling unit

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -481,6 +481,36 @@
 		legcuffed = null
 		update_inv_legcuffed(1)
 
+/mob/living/carbon/proc/get_temperature(var/datum/gas_mixture/environment)
+	var/loc_temp = T0C
+	if(istype(loc, /obj/mecha))
+		var/obj/mecha/M = loc
+		loc_temp =  M.return_temperature()
+
+	if(istype(loc, /obj/spacepod))
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/carbon.dm]: else if, not if

**Comment by PJB3005** [comment]
soooo, cold space = vox advantages are rip?

**Comment by Fox-McCloud** [comment]
@PJB3005 

Diona and IPCs never get cold enough to take damage; only cold--I'll do more testing, but I'm pretty sure Vox fall in this category as well. Also, Vox have access to EVA suits and they additionally have their own conversion kits to make hardsuits fit them.

**Comment by PJB3005** [comment]
Oh, okay, also, is it just me or did were those vox conversion kits never on the map?

**Comment by Markolie** [comment]
They were added a few weeks back.

**Comment by PJB3005** [comment]
shows how much I've played on paradise in the last month

**Comment by Fox-McCloud** [comment]
@PJB3005 

Any conversion kit vendor has Vox kits in them now; there's kits in engineering, near atmos, one in sec, and one in EVA.

Need to figure out why rooms are not cooling when exposed to space, though this may be a feature we do not want until LINDA makes its debut.

**Comment by PJB3005** [comment]
I bet it's related to ZAS, the same issue where areas can be superheated, even if exposed to space, because ZAS doesn't move the temperature around in that case

File: code/modules/mob/living/carbon/species.dm [hunk 2]
```diff
@@ -202,12 +195,16 @@
...
 				if(prob(20))
 					spawn(0) H.emote(pick("giggle", "laugh"))
-			SA.moles = 0
 
+	handle_temperature(breath, H)
+
+	return 1
+
+/datum/species/proc/handle_temperature(datum/gas_mixture/breath, var/mob/living/carbon/human/H) // called by human/life, handles temperatures
 	if( (abs(310.15 - breath.temperature) > 50) && !(RESIST_HEAT in H.mutations)) // Hot air hurts :(
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/species.dm]: Not sure if this should be 310.15 or species.body_temperature; doesn't really impact anyone other than Diona, so I'm going to guess this should be left alone.

**Comment by fj45** [comment]
Great change!

Just a couple of things I am a bit unsure about:

"N2O is easier to detect sooner"

Isn't this the biggest reason that you would use N2O to sabotage the station? Seems like using CO2 would just be far more effective. 

"This also removes some of the more annoying effects of plasma"

The clothing getting contaminated thing never really worked well but its kinda of shame to loose the other stuff, really made it clear how damned deadly the stuff is.

**Comment by Fox-McCloud** [comment]
@fj45

You'll start giggling a hair bit sooner, but the impacts of you sleeping should last longer so--it's a case of "easier to detect; hits harder once it does impact you"

Plasma effects are...strange; basically since the plasma concentration to trigger TOX is a lot lower, it means that you're really moving the dangers of plasma more into the gas itself rather than contamination (which causes tox based on some code specifically for that), burns your eyes and can even burn your skin+cause mutations (IF it's configured to do so, which I don't believe it is): https://github.com/ParadiseSS13/Paradise/blob/0128878100534d1d2c6b36afd0596772972777a5/code/ZAS/Plasma.dm#L77

Either case, I need to figure out why heat exchangers still suck, even with this change, though that may be best served for a different PR, TBH

**Comment by Fox-McCloud** [comment]
Update: Yeah, our heat exchanger code is vasttttly different than TG's so that's probably why they absolutely suck at cooling things off, independent of how hot/cold space is; it looks like we're going to have to have LINDA to have super cold space for *atmos*.

**Comment by fj45** [comment]
"Update: Yeah, our heat exchanger code is vasttttly different than TG's so that's probably why they absolutely suck at cooling things off, independent of how hot/cold space is; it looks like we're going to have to have LINDA to have super cold space for atmos."

Daaamn you baaay!

**Comment by Fox-McCloud** [comment]
Still not sure if I like this PR or not.

**Comment by Fox-McCloud** [comment]
Not much else needs to be changed, I don't think; from what I can tell species temperature and presure damages work (breathing and environmental). May be a few more things I need to technically test (cryo+mechs+spacedpods+transit tubes), but nothing major.

Would still like this reviewed by another pair of eyes.

**Comment by Earthdivine** [comment]
Considering the changes to Vox and needing suits, the Captain suit can't be modified for Vox use, which I assume is because there is no equivalent suit for them yet, meaning a Vox Captain needs to use a regular EVA suit.

**Comment by Fox-McCloud** [comment]
Can confirm that cryo, space pods, mechs, and transit tubes all function well.

**Comment by Fox-McCloud** [comment]
Re-added requirement to have pressure protection on BOTH the head and body to get full pressure protection.

**Comment by fj45** [comment]
Awesome!

**Comment by Fox-McCloud** [comment]
Ok, for whatever reason Bay had a separate proc for being in a fire to make you take even MORE damage for being in it rather than it just being something handled by your body temperature or your breathing temperature.

In any event this has been commented out; fires should still be incredibly deadly, just not +20 damage per tick deadly anymore.

Inevitably, this means that machine people are nigh invincible to fires since this proc was commented out; as and adjustment, their body temperature level 2 and 3 damages have been lowered; they're still very temperature resistant, just not nearly invincible to it.

**Comment by Fox-McCloud** [comment]
Oki doke; what I thought was my mistake or a bug is just more Bay silliness; at this stage, I'd like a second pair of eyes to take a look at this section: https://github.com/ParadiseSS13/Paradise/pull/511/files#diff-2de2df982739a4fbdb54a4b2715d6f92R597 to make sure the temperature values look correct in accordance with this: https://github.com/tgstation/-tg-station/blob/f68fd02d262cd04eaf3cf0dc37bc1f1aa97afdff/code/modules/mob/living/carbon/human/life.dm#L146

If it is, then this PR is ready to go.

**Comment by Fox-McCloud** [comment]
@Markolie and @ZomgPonies 

Ready for merging; section in question was reviewed by someone else.

**Pull request merged**